verifier/verifier.go

```diff
@@ -0,0 +1,153 @@
+package verifier
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/openpubkey/openpubkey/pktoken"
+)
+
+type ProviderVerifier interface {
+	// Returns the OpenID provider issuer as seen in ID token e.g. "https://accounts.google.com"
+	Issuer() string
+	ProviderPublicKey(ctx context.Context, token []byte) (JSONWebKey, error)
+	VerifyProvider(ctx context.Context, pkt *pktoken.PKToken) error
+}
+
+type CosignerVerifier interface {
+	Issuer() string
+	Strict() bool // Whether or not a given cosigner MUST be present for successful verification
+	VerifyCosigner(ctx context.Context, pkt *pktoken.PKToken) error
+}
+
+type VerifierOpts func(*Verifier) error
+
+func WithCosignerVerifiers(verifiers ...*DefaultCosignerVerifier) VerifierOpts {
+	return func(v *Verifier) error {
+		for _, verifier := range verifiers {
+			fmt.Println(verifier.issuer)
```

Comment:
If we are checking for existing issuers in `AddProviderVerifiers` we should add that check here as well